### PR TITLE
Doc for Batchnorm

### PR DESCRIPTION
**Pull request by diyessi**
Documentation for the three batchnorm functions.

### CODE REVIEW

File: doc/sphinx/source/ops/batch_norm_inference.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+
+
+Inputs
+------
+
++---------------------+-------------------------+------------------------------+
+| Name                | Element Type            | Shape                        |
++=====================+=========================+==============================+
+| ``input``           | real                    | :math:`(\bullet, C, \ldots)` |
```
Comment (aprocter) [inline doc/sphinx/source/ops/batch_norm_inference.rst]: maybe change `real` to `any real type`

File: doc/sphinx/source/ops/batch_norm_training.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+
+
+Inputs
+------
+
++---------------------+-------------------------+------------------------------+
+| Name                | Element Type            | Shape                        |
++=====================+=========================+==============================+
+| ``input``           | real                    | :math:`(\bullet, C, \ldots)` |
```
Comment (aprocter) [inline doc/sphinx/source/ops/batch_norm_training.rst]: same comment on `real`

File: doc/sphinx/source/ops/batch_norm_training.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+channel being axis 1. For each position, there are :math:`C` channel values, 
+each normalized independently.
+
+Normalization of a channel sample is controlled by two values:
+
+*  the `batch_mean` :math:`\mu`, and
+   
+*  the `batch_variance` :math:`\sigma^2`; 
+
+and by two scaling attributes: :math:`\gamma` and :math:`\beta`.
```
Comment (aprocter) [inline doc/sphinx/source/ops/batch_norm_training.rst]: I think sometimes these get called scale and offset respectively?

File: doc/sphinx/source/ops/batch_norm_training_backprop.rst [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+   \mathtt{centered}_i &= \mathtt{input}_i - \mathtt{mean}\\
+   \mathtt{square}_i &= \mathtt{centered}_i^2\\
+   \mathtt{variance} &= \frac{\sum \mathtt{square}_i}{N}\\
+   \mathtt{invsqrt} &= \frac{1}{\sqrt{\mathtt{variance}+\epsilon}}\\
+   \mathtt{gmul} &= \texttt{gamma}\cdot \mathtt{invsqrt}\\
+   \mathtt{normed}_i &= \mathtt{centered}_i\mathtt{gmul}+\texttt{beta}
+
+Using the notation :math:`\overline{\texttt{name}}` for :math:`\texttt{name_delta}`
+and :math:`\overline{x} \leftarrow y`
+to mean the backprop value for :math:`\texttt{x_delta}` is a sum that includes :math:`y`.
```
Comment (indie) [inline doc/sphinx/source/ops/batch_norm_training_backprop.rst]: This seems like it needs a comma or reordering to make sense?  Possibly reword to 

> the backprop value for :math:`\texttt{x_delta}` is a sum that includes :math:`y` when using the notation :math:`\overline{\texttt{name}}` for :math:`\texttt{name_delta}`and :math:`\overline{x} \leftarrow y`

?
